### DISCUSSION TITLE
Entities with Areas different From Device Area

### DISCUSSION THREAD

**Discussion by kzaoaai** [jseidl/magic-areas#455] [Q&A] [answered]
I have the Aqara FP2 with zone detection exposed to HA via homekit. This sensor has multiple zone detection, so under that device I have entites with different Areas. However, Magic Areas is ignoring the entity area and showing all entities under the FP2 device area.
![image](https://github.com/user-attachments/assets/4bfbe8b7-127e-4053-8642-175151fd0d5b)

**Comment by kzaoaai** [comment, marked as answer]
Okay I figured out a workaround: Remove the device area, and the entity areas start to be respected by Magic Areas.

**Comment by jseidl** [comment]
Sorry @kzaoaai I've missed this message. Yes MA will first trust the device area, if there isn't one then it looks into the entity area.
